src/LanguageServer/Protocol.TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -500,21 +500,23 @@ private static string GetDocumentFilePathFromName(string documentName)
 
     private static LSP.DidChangeTextDocumentParams CreateDidChangeTextDocumentParams(
         Uri documentUri,
-        ImmutableArray<(LSP.Range Range, string Text)> changes)
+        ImmutableArray<(LSP.Range? Range, string Text)> changes)
     {
-        var changeEvents = changes.Select(change => new LSP.TextDocumentContentChangeEvent
+        var changeEvents = new List<LSP.SumType<LSP.TextDocumentContentChangeEvent, LSP.TextDocumentContentChangeFullReplacementEvent>>(changes.Length);
```

Comment:
array builder?

Reply:
I was just going the simplest route since this is test code. Went ahead and just switched directly over to use an array as it's not really any more complicated.